src/resources/ResourceSnapshots/ResourceSnapshots.ts

```diff
@@ -65,40 +66,31 @@ export default class ResourceSnapshots extends Resource {
      * @param {File} file The file containing the configuration.
      * @param {CreateFromFileOptions} options
      */
-    createFromFile(file: File, options: CreateFromFileOptions): Promise<ResourceSnapshotsModel>;
+    createFromFile(file: File, options?: CreateFromFileOptions) {
+        const computedOptions = {
+            developerNotes: options.developerNotes,
+            snapshotFileType: this.getSnapshotFileType(file),
+        };
+        const form = getFormData();
+        form.append('file', file);
+
+        return this.api.postForm<ResourceSnapshotsModel>(
+            this.buildPath(`${ResourceSnapshots.baseUrl}/file`, computedOptions),
+            form
+        );
+    }
 
     /**
-     * Creates a snapshot from a file buffer containing the configuration.
+     * Creates a snapshot from a buffer containing the configuration.
      *
-     * @param {Buffer} file The file containing the configuration.
+     * @param {Buffer} file The buffer containing the configuration.
      * @param {ResourceSnapshotSupportedFileTypes} fileType The type of the file containing the configuration.
      * @param {CreateFromFileOptions} options
      */
-    createFromFile(
-        file: Buffer,
-        fileType: ResourceSnapshotSupportedFileTypes,
-        options: CreateFromFileOptions
-    ): Promise<ResourceSnapshotsModel>;
-
-    createFromFile(
-        file: File | Buffer,
-        typeOrOptions: ResourceSnapshotSupportedFileTypes | CreateFromFileOptions,
-        options?: CreateFromFileOptions
-    ) {
-        let fileContent: File | string;
-        let computedOptions = {developerNotes: undefined, snapshotFileType: undefined};
-
-        if (Buffer.isBuffer(file)) {
-            fileContent = file.toString();
-            computedOptions = {developerNotes: options.developerNotes, snapshotFileType: typeOrOptions};
-        } else {
-            fileContent = file;
-            computedOptions.developerNotes = (typeOrOptions as CreateFromFileOptions).developerNotes;
-            computedOptions.snapshotFileType = this.getSnapshotFileType(file);
-        }
-
-        const form: FormData = getFormData();
-        form.append('file', fileContent);
+    createFromBuffer(file: Buffer, fileType: ResourceSnapshotSupportedFileTypes, options: CreateFromFileOptions) {
+        const computedOptions = {developerNotes: options.developerNotes, snapshotFileType: fileType};
+        const form = getFormData();
+        ((form as unknown) as FormDataNode).append('file', file, `snapshot.${fileType.toString().toLowerCase()}`);
```

Comment:
This cast to `FormDataNode` is a bit unsafe. 

It would blow up in an unexpected way if you were to call this in a browser scenario (even though you "should not use them in a browser"). Since you can't pass in a Buffer directly in a browser context, I assume ?


I think we should leverage type guards to differentiate the available "Form" object.

My suggestion would be to modify `getFormData()` to return `FormData | FormDataNode`. And to also have a couple of function like these:  
`function isFormDataNode(form: FormData | FormDataNode): form is FormDataNode`
and
`function isFormDataBrowser(form: FormData | FormDataNode): form is FormData`

Those function could look at properties unique for each type to differentiate, and thus you would be assured that you are dealing with the correct object before calling append on it.

Reply:
I understand, but `FormData` and `FormDataNode` cannot intersect, which will simply push back the error somewhere else in the code.

Also, the [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) API used by the APICore does not support FormDataNode type...

So we will need to add a manual cast at some point.

Reply:
Another suggestion would be to have a `getFormDataNode()` which you are assured will return the correct object/type.

Would that work ?

Reply:
> Another suggestion would be to have a `getFormDataNode()` which you are assured will return the correct object/type.
> 
> Would that work ?

You'd have to cast the FromDataNode to a FormData down the line in APICore tho.

> even though you "should not use them in a browser". Since you can't pass in a Buffer directly in a browser context, I assume ?

Correct, except if someone set up a `window.Buffer`.

Reply:
> You'd have to cast the FromDataNode to a `FormData` down the line in APICore tho.
Correct